bats/tests/k8s/traefik.bats

```diff
@@ -0,0 +1,69 @@
+# Test case 25 & 26
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start k8s' {
+    start_kubernetes
+    wait_for_apiserver
+}
+
+get_host() {
+    if is_windows; then
+        echo "127.0.0.1"
+    else
+        echo "localhost"
+    fi
+}
+
+assert_traefik_pods_are_down() {
+    run kubectl get --all-namespaces pods
+
+    if  [[ "$output" != *"connection refused"* ]] && \
+        [[ "$output" != *"No resources found"* ]] && \
+        [[ "$output" != *"ContainerCreating"* ]] && \
+        [[ "$output" != *"Pending"* ]] && \
+        [[ "$output" != *"Completed"* ]] && \
+        [[ "$output" != *"Terminating"* ]] && \
+        [[ "$output" != *"traefik"* ]]; then
+        exit 0
+    else
+        exit 1
+    fi
+}
+
+assert_traefik_pods_are_up() {
+    run kubectl get --all-namespaces pods
+
+    if [[ "$output" != *"connection refused"* ]] && \
+        [[ "$output" != *"No resources found"* ]] && \
+        [[ "$output" != *"ContainerCreating"* ]] && \
+        [[ "$output" != *"Pending"* ]] && \
+        [[ "$output" != *"Terminating"* ]] && \
+        [[ "$output" =~ "traefik" ]]; then
+        exit 0
+    else
+        exit 1
+    fi
+}
+
+@test 'disable traefik' {
+    run rdctl set --kubernetes.options.traefik=FALSE
+    wait_for_apiserver
```

Comment:
I think this doesn't actually do anything here because the apiserver is likely still running and hasn't started to restart yet. It only works reliably when you change the k8s version, because `wait_for_apiserver` does a version check.

But I guess it doesn't matter because your `assert_traefik_*` functions are dealing with it.

Reply:
After `rdctl set `, the `wait_for_apiserver` does wait till the app comes back up again, I mean I noticed the function call waits till the status bar (almost) goes away. I feel it's still okay to have an another check to first ensure the app comes back up after the `rdctl reset`.